pep-9999.rst

```diff
@@ -140,6 +139,26 @@ logically conflict with each other in the requirement: ``version``, ``url``,
 
 An empty requirement table ``{}`` places no restriction on the requirement.
 
+Environment markers
+*******************
+
+All of the :pep:`508` environment marker variables are allowed as keys in the
+requirement table, with underscores ``_`` replace with hyphens ``-``. The
+values are strings which consitute the remainder of the environment marker.
+
+As an example, for the environment marker ``python_version < "2.7"``, the
+following is included in the requirement table:
+
+.. code-block:: toml
+
+    python-version = "< '2.7'"
```

Comment:
Should we allow something like the following translation? (example is not real)

```
aiohttp (>= 3.6.2) ; ( python_version '~= 3.6.1' or python_version '>=3.8.1' ) and ( extra == 'http' or extra == 'client')
```

```toml
aiohttp = { version = ">= 3.6.2", python-version = ["~= 3.6.1", ">= 3.8.1"], extra = ["client", "http"] }
```

Reply:
The alternative is the more explicit form.

```toml
aiohttp = [
    { version = ">= 3.6.2", python-version = "~= 3.6.1", extra = "http" },
    { version = ">= 3.6.2", python-version = ">= 3.8.1", extra = "client" }
]
```

Reply:
Building a matrix is fairly esoteric for what's likely a small use-case. Allowing a list of values is another type the user has to learn, and makes the PEP even more complex. Throughout this PEP, I've tried to stick with always having one line for each dependency specification, and not using fancy programming. In my experience, this has caused less bugs.

I think for now let's stick with the explicit form and allow a matrix down the road.

Reply:
I do not mind really. Wanted to mention this as a potential option. I am not sure exactly how small a use case it is. We have had to deal with a few of these questions in poetry. I like the explicit approach personally, because it forces the user to think about what they really want.

I am guessing this also means you want to keep the `extra` as it is as well then? So no list, only single value. At least for now.